lib/config.js

```diff
@@ -184,150 +184,152 @@ function getLocalConfig(thisConfig, directory) {
  * @class Config
  * @param {Object} options Options to be passed in
  */
-function Config(options) {
-    options = options || {};
+class Config {
+    constructor(options) {
+        options = options || {};
 
-    this.ignore = options.ignore;
-    this.ignorePath = options.ignorePath;
-    this.cache = {};
-    this.parser = options.parser;
-    this.parserOptions = options.parserOptions || {};
+        this.ignore = options.ignore;
+        this.ignorePath = options.ignorePath;
+        this.cache = {};
+        this.parser = options.parser;
+        this.parserOptions = options.parserOptions || {};
 
-    this.baseConfig = options.baseConfig ? loadConfig(options.baseConfig) : { rules: {} };
+        this.baseConfig = options.baseConfig ? loadConfig(options.baseConfig) : { rules: {} };
 
-    this.useEslintrc = (options.useEslintrc !== false);
+        this.useEslintrc = (options.useEslintrc !== false);
 
-    this.env = (options.envs || []).reduce((envs, name) => {
-        envs[name] = true;
-        return envs;
-    }, {});
+        this.env = (options.envs || []).reduce((envs, name) => {
+            envs[ name ] = true;
+            return envs;
+        }, {});
 
-    /*
-     * Handle declared globals.
-     * For global variable foo, handle "foo:false" and "foo:true" to set
-     * whether global is writable.
-     * If user declares "foo", convert to "foo:false".
-     */
-    this.globals = (options.globals || []).reduce((globals, def) => {
-        const parts = def.split(":");
+        /*
+         * Handle declared globals.
+         * For global variable foo, handle "foo:false" and "foo:true" to set
+         * whether global is writable.
+         * If user declares "foo", convert to "foo:false".
+         */
+        this.globals = (options.globals || []).reduce((globals, def) => {
+            const parts = def.split(":");
 
-        globals[parts[0]] = (parts.length > 1 && parts[1] === "true");
+            globals[ parts[ 0 ] ] = (parts.length > 1 && parts[ 1 ] === "true");
```

Comment:
FYI, not-an-aardvark has a chore PR open to set `array-bracket-spacing` to "never", so you may wish to either revert the spacing changes here or find his PR and object to it.